api/v1/fluxinstance_types.go

```diff
@@ -147,6 +154,44 @@ type Kustomize struct {
 	Patches []kustomize.Patch `json:"patches,omitempty"`
 }
 
+type Sync struct {
+	// Interval is the time between syncs.
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ms|s|m|h))+$"
+	// +kubebuilder:default:="1m"
+	// +optional
+	Interval *metav1.Duration `json:"interval,omitempty"`
+
+	// Kind is the kind of the source.
+	// +kubebuilder:validation:Enum=OCIRepository;GitRepository;Bucket
+	// +required
+	Kind string `json:"kind"`
+
+	// URL is the source URL, can be a Git repository HTTP/S or SSH address,
+	// an OCI repository address or a Bucket endpoint.
+	// +required
+	URL string `json:"url"`
+
+	// Ref is the source reference, can be a Git ref name e.g. 'refs/heads/main',
+	// an OCI tag e.g. 'latest' or a bucket name e.g. 'flux'.
+	// +required
+	Ref string `json:"ref"`
```

Comment:
for an `OCIRepository` we are expecting user to use `flux push` from our discussion. Shouldn't we have this `optional`? And also use `LayerSelector`?